### PR TITLE
Notebook Markdown email rendering fix

### LINKED ISSUE #15983: Notebook email address becomes a broken email link
- Azure Data Studio Version:
Version: 1.30.0 (user setup)
Commit: 59c4b8e90cf2d3a24bed72623197f10f8d090fdc
Date: 2021-06-16T19:04:08.858Z
VS Code: 1.53.0
Electron: 11.2.2
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: Windows_NT x64 10.0.19043

Steps to Reproduce:

1. In a new notebook, use a markdown view or WYSIWYG view and type in the following in a new text cell:
`Link to myemail@email.com`
2. Exit out of the text cell. You'll see that the email address becomes a link. 
3. Click on the link and you'll see that it will try to link to mailto with the following address:
`&#109; &#x79; &#x65; &#109; &#x61; &#105; &#108; &#x40; &#101; &#x6d; &#97; &#x69; &#108; &#x2e; &#99; &#111; &#109`

See a quick video below:
WeirdEmailLink

Does this issue occur when all extensions are disabled?: Yes.

### PR DESCRIPTION
**Pull request by VasuBhog**
This PR fixes #15983. 

The issue occurs when the user directly types in the email address, as the parsing `test@email.com` would double encode the `&` when converting it to a proper `mailto` format. The resulting markdown that is resolved contains the `&amp` and upon clicking the link it would send the improper format to the email client:
image

In order to fix it, I removed the additional encoding (specifically for emails uri's) to make sure that the `&` is still encoded for other uri schemas.

### CODE REVIEW

**Review by barbaravaldez — APPROVED** [review]

**Review by Charles-Gagnon — APPROVED** [review]
It seems pretty strange for the href to be &-encoded already - do you know why that happens? I'm guessing it's about something to do with making sure reserved characters are rendered properly - but if we had a doc or something that said so definitively that would be handy I think to make sure we're handling this correctly.

File: src/sql/workbench/contrib/notebook/browser/outputs/notebookMarkdown.ts [hunk 1]
```diff
@@ -134,11 +134,19 @@ export class NotebookMarkdownRenderer {
 
 			} else {
 				// HTML Encode href
-				href = href.replace(/&(?!amp;)/g, '&amp;')
-					.replace(/</g, '&lt;')
-					.replace(/>/g, '&gt;')
-					.replace(/"/g, '&quot;')
-					.replace(/'/g, '&#39;');
+				let uri = URI.parse(href);
+				if (uri.scheme === 'mailto') {
```
[the remark this answers is not in the document]
Comment (VasuBhog) [reply src/sql/workbench/contrib/notebook/browser/outputs/notebookMarkdown.ts:138]: I did look for a doc, but the only way I was able to see this happening was after debugging and understand that the &amp get encoded and decoded via this place and marked js. As such we don't need to do the extra replacement here.

**Comment by Charles-Gagnon** [comment]
FYI @VasuBhog - it looks like this is controlled by the [mangle option](https://marked.js.org/using_advanced#options)

https://github.com/markedjs/marked/discussions/2142#discussioncomment-1044036

It would probably be good to look into whether setting that to false solves the issue as well since that would seem to me to be a better supported way of fixing this.

**Comment by VasuBhog** [comment]
> FYI @VasuBhog - it looks like this is controlled by the [mangle option](https://marked.js.org/using_advanced#options)
> 
> [markedjs/marked#2142 (comment)](https://github.com/markedjs/marked/discussions/2142#discussioncomment-1044036)
> 
> It would probably be good to look into whether setting that to false solves the issue as well since that would seem to me to be a better supported way of fixing this.

Will check it out!

**Pull request merged**